api/v1beta1/hyperconverged_types.go

```diff
@@ -387,6 +387,10 @@ type VirtualMachineOptions struct {
 // by default yet.
 // +k8s:openapi-gen=true
 type HyperConvergedFeatureGates struct {
+	// Allow to expose a limited set of host metrics to guests.
+	// +optional
+	DownwardMetrics *bool `json:"downwardMetrics,omitempty"`
```

Comment:
We need to understand why we want to do this:
with something like this the feature can only be fully on/fully off at cluster level.
If we think that having the DownwardMetrics always enabled can introduce some security concern, we should probably introduce something more fine grained to let the cluster admin selectively enable it just for trusted VMs/namespaces.

In this way, if just a single VM on the cluster need this for a valid reason, this could only be enabled for all the VMs on the cluster and so probably we are not reaching what we want to offer to the cluster admin.

Reply:
I agree. But as first step to go there, I think we should follow this path. WDYT?

Reply:
DownwardMetrics is always on since version 1.5 (2.5 years ago) so I don't see any urgency to introduce a temporary API and then to deprecate it later on in order to move to a properly designed one.
If we already tolerated it for 2.5 years, I think we can wait for a few days/weeks more to get a proper API.